json.md

```diff
@@ -0,0 +1,301 @@
+/// source generator to generate JSON serialization logic.
+[[highlight]]@JsonSerializable()[[/highlight]]
+
+/// Every json_serializable class must have the serializer mixin. 
+/// It makes the generated toJson() method to be usable for the class. 
+/// The mixin's name follows the source class, in this case, User.
+class User extends Object with _$[[highlight]]User[[/highlight]]SerializerMixin {
+  User(this.name, this.email);
+
+  String name;
+  String email;
+
+  /// A necessary factory constructor for creating a new User instance
+  /// from a map. We pass the map to the generated _$UserFromJson constructor. 
+  /// The constructor is named after the source class, in this case User.
+  factory User.fromJson(Map<String, dynamic> json) => _$[[highlight]]User[[/highlight]]FromJson(json);
+}
+{% endprettify %}
+
+To deserialize a JSON string `json_serializable` way, we do not have actually to make any changes to our previous code.
+
+```dart
+Map map = JSON.decode(json);
+User user = new User.fromJson(map);
+```
+
+Same goes for serialization. The calling API is the same as before.
+
+```dart
+String json = JSON.encode(user.toJson());
+```
+
+With `json_serializable`, we can forget any manual JSON serialization in the `User` class. The source code generator creates a file called `user.g.dart`, which has all the necessary serialization logic. Now we do not necessarily have to write automated tests to be sure that the serialization works - it is now _the library's responsibility_ to make sure the serialization works appropriately.
+
+Now that we have learned how the model classes look, let's go through the required steps to introduce `json_serializable` to our project.
+
+### Setting up json_serializable in a project
+
+To have `json_serializable` up and running, we have to go through some steps first. We need to do this only ** once per project**. After the initial setup, creating JSON model classes is trivial. 
+
+#### Setting up the pubspec dependencies
+
+To include `json_serializable` in our project, we need one regular and two _dev\ dependencies_. In short, _dev dependencies_ are dependencies that are not included in our app source code.
+
+The latest versions of these required dependencies can be seen by following [this link](https://github.com/dart-lang/json_serializable/blob/master/example/pubspec.yaml).
+
+**pubspec.yaml**
+
+```yaml
+dependencies:
+  # Your other regular dependencies here
+  json_annotation: ^0.2.2
+
+dev_dependencies:
+  # Your other dev_dependencies here
+  build_runner: ^0.6.1
+  json_serializable: ^0.3.0
+```
+
+After declaring the needed dependencies, synchronize your packages. In IntelliJ IDEA, click the _Packages get_ link on the top of an opened `pubspec.yaml` file. Alternatively, you can do synchronize packages by running `flutter packages get` from the command line in your projects' root folder.
+
+Next,  define some build actions.
+
+#### Defining the build actions
```

Comment:
FYI – this has been simplified a lot.

See https://github.com/dart-lang/json_serializable/tree/master/example

Reply:
Great! I'll look into it and update this section.

Reply:
Awesome.

Reply:
I tried to take a quick look into it, and ran into the exact same issue that [this reddit user describes in this comment](https://www.reddit.com/r/FlutterDev/comments/7nymf5/simple_json_with_flutter/ds80p88/).

This is reproducible on the latest Flutter version, and happens by simply following these steps:

* create a new Flutter project
* add the dependencies listed on the `json_serializable` README
* attempt to get the packages -> a long error message which states that a version `^0.7.0` for `build_runner` was found.

Should I create a more detailed issue report, or is this some common issue that can be resolved easily?

Reply:
We're in dependency lock, I'm pretty sure.

I think I have a path to untangle the knot (at least a bit). Follow here: https://github.com/dart-lang/json_serializable/issues/95